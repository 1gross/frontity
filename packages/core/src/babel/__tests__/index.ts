import getBabel from '..';

test('Babel returns the moduleDev config', () => {
  expect(getBabel('moduleDev')).toMatchSnapshot();
});

test('Babel returns the moduleProd config', () => {
  expect(getBabel('moduleProd')).toMatchSnapshot();
});

test('Babel returns the es5 config', () => {
  expect(getBabel('es5')).toMatchSnapshot();
});

test('Babel returns the node config', () => {
  expect(getBabel('node')).toMatchSnapshot();
});
