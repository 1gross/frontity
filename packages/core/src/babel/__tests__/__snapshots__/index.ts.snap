// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel returns the es5 config 1`] = `
Object {
  "plugins": Array [
    "transform-object-rest-spread",
    "transform-class-properties",
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "targets": Object {
          "browsers": Array [
            "and_chr >= 67",
            "and_ff >= 18",
            "and_uc >= 11.8",
            "android >= 67",
            "not android <= 4.4.4",
            "chrome >= 49",
            "edge >= 12",
            "firefox >= 18",
            "ios_saf >= 10",
            "not op_mini all",
            "op_mob >= 46",
            "opera >= 36",
            "safari >= 10",
            "samsung >= 5",
          ],
        },
        "useBuiltIns": "usage",
      },
    ],
    "react",
  ],
}
`;

exports[`Babel returns the moduleDev config 1`] = `
Object {
  "plugins": Array [
    "transform-object-rest-spread",
    "transform-class-properties",
    "react-hot-loader/babel",
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "targets": Object {
          "esmodules": true,
        },
        "useBuiltIns": "usage",
      },
    ],
    "react",
  ],
}
`;

exports[`Babel returns the moduleProd config 1`] = `
Object {
  "plugins": Array [
    "transform-object-rest-spread",
    "transform-class-properties",
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "targets": Object {
          "esmodules": true,
        },
        "useBuiltIns": "usage",
      },
    ],
    "react",
  ],
}
`;

exports[`Babel returns the node config 1`] = `
Object {
  "plugins": Array [
    "transform-object-rest-spread",
    "transform-class-properties",
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "targets": Object {
          "node": "8.10",
        },
        "useBuiltIns": "usage",
      },
    ],
    "react",
  ],
}
`;
